packages/legend-extension-dsl-data-space/src/stores/query/DataSpaceTemplateQueryPromotionReviewerStore.ts

```diff
@@ -0,0 +1,666 @@
+        .filter(
+          (workspace) =>
+            // NOTE we don't handle workspaces that only have conflict resolution but no standard workspace
+            // since that indicates bad state of the SDLC server
+            !workspacesInConflictResolutionIds.includes(
+              workspace.workspaceId,
+            ) && workspace.workspaceType === WorkspaceType.GROUP,
+        );
+      this.loadWorkspacesState.pass();
+    } catch (error) {
+      assertErrorThrown(error);
+      this.applicationStore.logService.error(
+        LogEvent.create(LEGEND_STUDIO_APP_EVENT.SDLC_MANAGER_FAILURE),
+        error,
+      );
+      this.applicationStore.notificationService.notifyError(error);
+      this.loadWorkspacesState.fail();
+    }
+  }
+
+  *promoteAsTemplateQuery(): GeneratorFn<void> {
+    const query = this.currentQuery;
+    const project = this.currentProject;
+    if (
+      this.promoteState.isInProgress ||
+      !query ||
+      !this.currentQueryInfo ||
+      !this.currentProjectConfiguration ||
+      !project ||
+      !this.workspaceName ||
+      !this.templateQueryTitle ||
+      !this.dataSpace ||
+      !this.isWorkspaceNameValid ||
+      !this.isTemplateQueryIdValid
+    ) {
+      return;
+    }
+
+    try {
+      this.promoteState.inProgress();
+      // 1. prepare project entities
+      this.applicationStore.alertService.setBlockingAlert({
+        message: `Fetching and updating project...`,
+        prompt: 'Please do not close the application',
+        showLoading: true,
+      });
+      const dataSpaceExecutableTemplate = new DataSpaceExecutableTemplate();
+      dataSpaceExecutableTemplate.id = this.templateQueryId;
+      dataSpaceExecutableTemplate.title = this.templateQueryTitle;
+      dataSpaceExecutableTemplate.description = this.templateQueryDescription;
+      dataSpaceExecutableTemplate.executionContextKey = guaranteeNonNullable(
+        this.dataSpace.executionContexts.filter(
+          (ec) =>
+            ec.mapping.value.path ===
+              guaranteeNonNullable(this.currentQueryInfo).mapping &&
+            ec.defaultRuntime.value.path ===
+              guaranteeNonNullable(this.currentQueryInfo).runtime,
+        )[0]?.name,
+        'can`t find a corresponding executatin key based on query`s mapping and runtime in dataspace',
+      );
+      dataSpaceExecutableTemplate.query =
+        (yield this.graphManagerState.graphManager.pureCodeToLambda(
+          this.currentQueryInfo.content,
+        )) as RawLambda;
+      this.dataSpace.executables = this.dataSpace.executables
+        ? [...this.dataSpace.executables, dataSpaceExecutableTemplate]
+        : [dataSpaceExecutableTemplate];
+
+      // 2. check if the graph compiles properly
+      this.applicationStore.alertService.setBlockingAlert({
+        message: `Checking workspace compilation status...`,
+        prompt: 'Please do not close the application',
+        showLoading: true,
+      });
+
+      const graphGrammar =
```

Comment:
lets avoid the studio graph here. we can add methods to pure graph manager if need be. 
but we should be able to do patch the dataspace and add the template query without building the graph. 
We can still compile with engine with the full patched entities but we shouldnt need to build our graph for this.